MediaGalleryUi/Ui/Component/Listing/Columns/Url.php

```diff
@@ -24,23 +24,49 @@ class Url extends Column
     private $storeManager;
 
     /**
+     * @var UrlInterface
+     */
+    private $url;
```

Comment:
This column should not be aware of directory tree. A dedicated `class` can be assigned to the directory tree container in the xml to prepare the URL